src/ucp/tag/rndv.c

```diff
@@ -715,15 +709,17 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr),
             }
         }
         /* put protocol is allowed - register receive buffer memory for rma */
+        ucs_assert(rndv_rts_hdr->size <= rreq->recv.length);
         ucp_request_recv_buffer_reg(rreq, ucp_ep_config(ep)->key.rma_bw_md_map,
-                                    ucs_min(rreq->recv.length, rndv_rts_hdr->size));
+                                    rndv_rts_hdr->size);
```

Comment:
should do ucs_min

Reply:
Since we already checked it on 676 line:
` if (ucs_unlikely(rreq->recv.length < rndv_rts_hdr->size)) {`

it's guaranteed that `rndv_rts_hdr->size <= rreq->recv.length` and we can directly use `rndv_rts_hdr->size` here

Reply:
ok